### PR TITLE
get_MOT()

### PR DESCRIPTION
**Pull request by wengraf**
I've tested this with a range of lists of VRMs and it seems to work as it should.  I've included some code to protect against failure with specific vehicles with incomplete data (e.g., try()).

This is my first bit of function writing to be public, so be kind!

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -49,7 +49,9 @@ Suggests:
     leaflet,
     geojsonsf,
     htmltools,
-    tmap
+    tmap,
+    jsonlite,
+    utils
```
Comment (Robinlovelace) [inline DESCRIPTION]: utils is a base R package so this is not needed I think.

File: R/mot.R [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+#' @param VRMs A list of VRMs as character strings.
+#' @param apikey Your API key as a character string.
+#'
+#' @export
+#' @examples
+#' \donttest{
+#' get_MOT_STATS19(VRMs = VRMs, apikey = apikey)
+#' }
+
+get_MOT_STATS19 <- function(VRMs, apikey) {
```
Comment (Robinlovelace) [inline R/mot.R]: `=` assignment pls for consistency.

**Comment by Robinlovelace** [comment]
Update, seems Travis checks are failing because there is no VRMs object in the example/test code: https://travis-ci.org/ropensci/stats19/jobs/602819774#L1203

File: DESCRIPTION (hunk 1 above)
Comment (wengraf) [reply DESCRIPTION]: Sorry, I put utils in because the check was throwing up a note otherwise.

File: R/mot.R (hunk 2 above)
Comment (wengraf) [reply R/mot.R]: OK - do you want me to change this or should you?

**Comment by wengraf** [comment]
OK, I add an example VRM in.

**Comment by Robinlovelace** [comment]
Aha, no I think you just need to put `utils::` before the function that is the culprit. See here for example: https://github.com/ropensci/stplanr/blob/4e2f37187577bfb7b6c2f17dfd97bbf03ff725af/R/google-functions.R#L132

**Comment by wengraf** [comment]
OK.  I've tried to sort out all the issues mentioned.

**Comment by mem48** [comment]
Hi @wengraf some first thoughts.

1.  It would be good to default the api key to something like `apikey = Sys.getenv("MOT_key")` asi it makes it easy for the users and avoids publishing the key in code.

1. Why do `VRMs` need to be a list? a character vector would be easier for the user?

1. I suggest vrm not VRMs which is harder to type

1. Look at using the httr package which makes using curl request much easer and concise

1. You seem to be mixing data retrieval and analysis it is best practice to do these separately

**Comment by Robinlovelace** [comment]
> It would be good to default the api key to something like apikey = Sys.getenv("MOT_key") asi it makes it easy for the users and avoids publishing the key in code.

True that, can add post merge though as possibly quicker for us than @wengraf 

> Why do VRMs need to be a list? a character vector would be easier for the user?

Agreed. Same with upper case in function names, all other functions are all lowercase.

> Look at using the httr package which makes using curl request much easer and concise

Can you be more specific or put in a PR @mem48 ?

> You seem to be mixing data retrieval and analysis it is best practice to do these separately

Again being specific or better putting in a PR will help clarify exactly what you mean.

Thanks for the comments though, very useful + will help get this in there asap.

**Comment by wengraf** [comment]
To address @mem48 's points:

1. Agree with both @mem48 and @Robinlovelace .
2 & 3. Agree - I'll change.
4.. I'm not comfortable in switching to a different package - if you can get a more concise package to work, fine.  But I'm not changing it myself.
5. I'd agree with you in a pure MOT package, but this is a function whose aim is to influence the discussion on DVLA/DVSA data provision, rather than provide long-term general utility.  The advantage of a one-liner demonstration (for a function that requires both CRASH access _and_ a DVSA account to work, so a pretty niche group of package users) outweighs the argument for the clarity of splitting the function into data retrieval/data analysis.  I imagine in an MOT package, you'd have one data retrieval function for each of the API endpoints, then topic specific analysis functions (e.g., vehicle characteristics, safety history, mileage estimates, etc.).

I'll get the kettle on and have another go now.

**Comment by Robinlovelace** [comment]
And in terms of making Travis happy, I think there's definitely an error in the code documented here: https://travis-ci.org/ropensci/stats19/builds/602876737#L1133

@wengraf you can test the package passes tests locally by pressing `At+Shift+E`.

**Comment by Robinlovelace** [comment]
More info: https://github.com/rstudio/rstudio/issues/3464

**Comment by mem48** [comment]
@wengraf I'm happy with your responces, my comments very much just the first things that came to mind. In this case I think speed is favoured over code elegance. But a proper MOT package will require a deeper rewrite.

**Comment by Robinlovelace** [comment]
I'll have a look now and try to get the tests passing.

**Comment by Robinlovelace** [comment]
Heads-up @wengraf the most recent change here should fix the travis fails: https://github.com/wengraf/stats19/pull/1 you should be able to merge this into your branch, will make the two branches in-line, except for your changes.

**Comment by wengraf** [comment]
> @wengraf I'm happy with your responces, my comments very much just the first things that came to mind. In this case I think speed is favoured over code elegance. But a proper MOT package will require a deeper rewrite.

Absolutely, @mem48  - I agree entirely.

**Comment by Robinlovelace** [comment]
Travis is passing 🎉

**Comment by wengraf** [comment]
Brilliant - thanks very much for letting me participate.

**Pull request merged**